.gitignore

```diff
@@ -1 +1,41 @@
+# Clang code formatting file
```

Comment:
Is it really necessary to add a .gitignore? I'm not sure we really want this in the repo, especially since some of the files are OSX specific...